client/js/table-view.js

```diff
@@ -1,88 +1,214 @@
 const { removeChildren, createTH, createTR, createTD } = require('./dom-util');
-
+ 
 class TableView {
     constructor(model) {
         this.model = model;
     }
-
+    // Initalizes refences, location, table, and events
     init() {
         this.initDomReferences();
         this.initCurrentCell();
         this.renderTable();
         this.attachEventHandlers();
     }
 
+    //initalizes references for later use 
     initDomReferences() {
         this.headerRowEl = document.querySelector('THEAD TR');
         this.sheetBodyEl = document.querySelector('TBODY');
-        this.formulaBarEl = document.querySelector('#formula-bar')
+        this.formulaBarEl = document.querySelector('#formula-bar');
+        this.sumBar = document.querySelector('#sum-bar');
+        this.columnButton = document.querySelector('#addColumn');
+        this.rowButton = document.querySelector('#addRow');
     }
 
+    // defines location at program start
     initCurrentCell() {
         this.currentCellLocation = { col: 0, row: 0 };
         this.renderFormulaBar();
     }
 
+    // prevents undefined from appearing in formula bar
     normalizeValueForRendering(value) {
         return value || '';
     }
 
+    // Renders the formula bar from cell location and value
     renderFormulaBar() {
         const currentCellValue = this.model.getValue(this.currentCellLocation);
         this.formulaBarEl.value = this.normalizeValueForRendering(currentCellValue);
         this.formulaBarEl.focus();
     }
+    //Calls to renders sections of the table
     renderTable() {
         this.renderTableHeader();
         this.renderTableBody();
+        this.renderSumBar();
     }
+
+    // Renders A, B, C, ... through ZZ for table header.
     renderTableHeader() {
         removeChildren(this.headerRowEl);
+        this.headerRowEl.appendChild(createTH(' '))
         getLetterRange('A', this.model.numCols)
             .map(colLabel => createTH(colLabel))
             .forEach(th => this.headerRowEl.appendChild(th));
     }
 
+    // Renders the sum of all the columns
+    renderSumBar() {
+        const sumData = new Array(this.model.numCols);
+        for (let col = 0; col < this.model.numCols + 1; col++) {
+            sumData[col] = 0;
+            for (let row = 0; row < this.model.numRows; row++) {
+                const position = { col: col - 1, row: row };
+                const value = parseInt(this.model.getValue(position))
+                if ((typeof value === 'number') && !isNaN(value)) {
+                    sumData[col] = sumData[col] + value;
+                }
+            }
+        }
+        removeChildren(this.sumBar);
+        sumData.map(colSum => createTD(colSum)).forEach(cs => this.sumBar.appendChild(cs));
+    }
+
+    // returns if the given location (in col, row) matches actual location  
     isCurrentCell(col, row) {
         return this.currentCellLocation.col === col &&
             this.currentCellLocation.row === row;
```

Comment:
Even better if you can line up the "this.currentCellLocation.row" just underneath its brother above. Aligning very similar lines helps a lot with readability, and looks really pleasing (something that contributes to people thinking your code is excellent).